includes/wcpdf-functions.php

```diff
@@ -169,14 +169,18 @@ function wcpdf_deprecated_function( $function, $version, $replacement = null ) {
 	if ( apply_filters( 'wcpdf_disable_deprecation_notices', false ) ) {
 		return;
 	}
+	
 	// if the deprecated function is called from one of our filters, $this should be $document.
-	$filter = current_filter();
+	$filter               = current_filter();
 	$global_wcpdf_filters = array( 'wp_ajax_generate_wpo_wcpdf' );
-	if ( ! in_array( $filter, $global_wcpdf_filters ) && strpos( $filter, 'wpo_wcpdf' ) !== false && strpos( $replacement, '$this' ) !== false ) {
-		$replacement = str_replace( '$this', '$document', $replacement );
+	
+	if ( ! empty( $filter ) && ! empty( $replacement ) && ! in_array( $filter, $global_wcpdf_filters ) && false !== strpos( $filter, 'wpo_wcpdf' ) && false !== strpos( $replacement, '$this' ) ) {
+		$replacement = ! empty( $replacement ) ? str_replace( '$this', '$document', $replacement ) : $replacement;
```

Comment:
It seems like you checked the emptiness of `$replacement` twice since you did in the if condition.